src/main/java/edu/harvard/iq/dataverse/util/json/InAppNotificationsJsonPrinter.java

```diff
@@ -0,0 +1,249 @@
+    public static final String KEY_REQUESTOR_LAST_NAME = "requestorLastName";
+    public static final String KEY_REQUESTOR_EMAIL = "requestorEmail";
+    public static final String KEY_DATAFILE_ID = "dataFileId";
+    public static final String KEY_DATAFILE_DISPLAY_NAME = "dataFileDisplayName";
+    public static final String KEY_ROOT_DATAVERSE_NAME = "rootDataverseName";
+    public static final String KEY_GUIDES_BASE_URL = "userGuidesBaseUrl";
+    public static final String KEY_GUIDES_VERSION = "userGuidesVersion";
+    public static final String KEY_GUIDES_SECTION_PATH = "userGuidesSectionPath";
+    public static final String KEY_CURATION_STATUS = "currentCurationStatus";
+    public static final String KEY_ADDITIONAL_INFO = "additionalInfo";
+
+    public static final String GUIDES_SECTION_PATH_DATAVERSE_MANAGEMENT_HTML = "user/dataverse-management.html";
+    public static final String GUIDES_SECTION_PATH_DATASET_MANAGEMENT_HTML = "user/dataset-management.html";
+    public static final String GUIDES_SECTION_PATH_USER_HTML = "user/index.html";
+
+    @EJB
+    private DataverseServiceBean dataverseService;
+    @EJB
+    private DatasetServiceBean datasetService;
+    @EJB
+    private DatasetVersionServiceBean datasetVersionService;
+    @EJB
+    private DataFileServiceBean dataFileService;
+    @EJB
+    private PermissionServiceBean permissionService;
+    @EJB
+    private SystemConfig systemConfig;
+
+    /**
+     * Populates a JSON builder with fields specific to the notification type.
+     *
+     * @param notificationJson  The JSON builder to add fields to.
+     * @param authenticatedUser The user receiving the notification.
+     * @param userNotification  The notification object containing the details.
+     */
+    public void addFieldsByType(final NullSafeJsonBuilder notificationJson, final AuthenticatedUser authenticatedUser, final UserNotification userNotification) {
+        final AuthenticatedUser requestor = userNotification.getRequestor();
+
+        switch (userNotification.getType()) {
+            case ASSIGNROLE:
+            case REVOKEROLE:
+                addRoleFields(notificationJson, authenticatedUser, userNotification);
+                break;
+            case CREATEDV:
+                addCreateDataverseFields(notificationJson, userNotification);
+                break;
+            case REQUESTFILEACCESS:
+                addRequestFileAccessFields(notificationJson, userNotification, requestor);
+                break;
+            case REQUESTEDFILEACCESS:
+            case GRANTFILEACCESS:
+            case REJECTFILEACCESS:
+                addDataFileFields(notificationJson, userNotification);
+                break;
+            case DATASETCREATED:
+                addDatasetCreatedFields(notificationJson, userNotification, requestor);
+                break;
+            case CREATEDS:
+                addCreateDatasetFields(notificationJson, userNotification);
+                break;
```

Comment:
Did we ever figure out anything about `CREATEDS` vs `DATASETCREATED`. Can we add some Javadoc somewhere about the difference?